modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/parser/Node.java

```diff
@@ -103,13 +103,18 @@ else if (matchLength == uriFragment.length()) {
     }
 
     public ResourceInfo getResource(Map<String, String> requestDetails) {
-        for (ResourceInfo resourceInfo : this.resource) {
-            if (resourceInfo.getAnnotationAttachmentInfo(Constants.HTTP_PACKAGE_PATH,
-                    requestDetails.get(Constants.HTTP_METHOD)) != null) {
-                return resourceInfo;
+        try {
+            for (ResourceInfo resourceInfo : this.resource) {
+                if (resourceInfo.getAnnotationAttachmentInfo(Constants.HTTP_PACKAGE_PATH,
+                        requestDetails.get(Constants.HTTP_METHOD)) != null) {
+                    return resourceInfo;
+                }
             }
+            return tryMatchingToDefaultVerb(requestDetails);
+        } catch(Throwable e) {
```

Comment:
yes, this.resource can be null. I experiences it when the request doesn't contains a resource.  Eg. http://localhost:9090/echo